metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java

```diff
@@ -130,4 +146,86 @@ public ResponseEntity<String> getIndexSizes() {
             .collect(Collectors.toList()));
     return ResponseEntity.ok(j.toString());
   }
+
+  @Tag(name = "ElasticSearchOperations")
+  @GetMapping(path = "/explainSearchQuery", produces = MediaType.APPLICATION_JSON_VALUE)
+  @Operation(summary = "Explain Search Query")
+  public ResponseEntity<ExplainResponse> explainSearchQuery(
+      @Parameter(
+              name = "query",
+              required = true,
+              description =
+                  "Query to evaluate for specified document, will be applied as an input string to standard search query builder.")
+          @RequestParam("query")
+          @Nonnull
+          String query,
+      @Parameter(
+              name = "documentId",
+              required = true,
+              description = "Document ID to apply explain to.")
+          @RequestParam("documentId")
+          @Nonnull
+          String documentId,
+      @Parameter(
+              name = "entityName",
+              required = true,
+              description = "Name of the entity the document belongs to.")
+          @RequestParam("entityName")
+          @Nonnull
+          String entityName,
+      @Parameter(name = "from", required = true, description = "Start point for pagination.")
```

Comment:
I guess this is used, but not sure why we're not using the scroll api

Reply:
We can switch it over to scroll if you think that'd be more useful, scroll is still not widely used for search though.